### PR TITLE
Fix: google drive incremental workflow stop using cron

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Refactor Google incremental workflow to stop using cron as it doesn't work when we "continueAsNew" within the workflow.
Instead, sleep and start again.

## Risk

Break the gdrive workflows.

## Deploy Plan

Deploy connectors, `restart-all-incremental-sync-workflows`

### CODE REVIEW

**Review by fontanierh — APPROVED** [review]
LGTM

File: connectors/src/connectors/google_drive/temporal/workflows.ts [hunk 1]
```diff
@@ -238,6 +235,9 @@ export async function googleDriveIncrementalSync(
   }
 
   await syncSucceeded(connectorId);
+
+  await sleep("60 minutes");
```
Comment (spolu) [inline connectors/src/connectors/google_drive/temporal/workflows.ts:239]: Hmmmmm so you want to give up entirely on cron?

I presume the motivation is that continue as new and are not compatible but we can't really remove the continue as new here?

A solution would be to cap the amount of work done in these incremental syncs but I reckon it's more work.

The big change here is that these workflows can't be cancelled anymore which should be documented in the associated run book + publicized to the team

Comment (Fraggle) [reply connectors/src/connectors/google_drive/temporal/workflows.ts:239]: > I presume the motivation is that continue as new and are not compatible but we can't really remove the continue as new here?

> A solution would be to cap the amount of work done in these incremental syncs but I reckon it's more work.

We do use continueAsNew extensively.
In term of alternative, there is also the "new" schedule feature (but it seemed a too significant change to me to do it in this PR).

> The big change here is that these workflows can't be cancelled anymore which should be documented in the associated run book + publicized to the team

Indeed, but FYI, this is the existing pattern of [Notion](https://github.com/dust-tt/dust/blob/5d8f04950a1bb9aff1caf938e7d24fdfcd089383/connectors/src/connectors/notion/temporal/workflows/index.ts#L217) & [Microsoft](https://github.com/dust-tt/dust/blob/e324fad4745eab071ff705dbdc3880d0b3e609a3/connectors/src/connectors/microsoft/temporal/workflows.ts#L180). I'm not sure anyone expect a cancelled workflow to come back to life automatically (and we have the production check catching them).

I'm also not sure that a cancelled cron workflow would have magically re-vive either.

If we want to be able to cancel a workflow and yet have the "next" run to work, I think [Schedule](https://temporal.io/blog/temporal-schedules-reliable-scalable-and-more-flexible-than-cron-jobs) is the correct way.

Comment (fontanierh) [reply connectors/src/connectors/google_drive/temporal/workflows.ts:239]: > The big change here is that these workflows can't be cancelled anymore which should be documented in the associated run book + publicized to the team

are you sure cron workflows can be cancelled ? I think it's exactly the same as with a timer, if you cancel it's gone

Comment (fontanierh) [reply connectors/src/connectors/google_drive/temporal/workflows.ts:239]: IMHO the advantages of using CRONs are really unclear

Comment (spolu) [reply connectors/src/connectors/google_drive/temporal/workflows.ts:239]: Roger that! Thanks for the context!

Comment (Fraggle) [reply connectors/src/connectors/google_drive/temporal/workflows.ts:239]: You're welcome !

**Pull request merged**